src/backend/distributed/planner/multi_logical_optimizer.c

```diff
@@ -2907,14 +2907,15 @@ IsPartitionColumnRecursive(Expr *columnExpression, Query *query)
 	List *rangetableList = query->rtable;
 	Index rangeTableEntryIndex = 0;
 	RangeTblEntry *rangeTableEntry = NULL;
+	Expr *strippedColumnExpression = strip_implicit_coercions(columnExpression);
```

Comment:
Ugh. We need to enable -Werror in Travis. Looked OK to me but I didn't inspect warnings (I've become used to the ones we have).

Can you slap some casts in here? That's all it'll take, right?